core/src/main/java/org/apache/gravitino/listener/api/event/AlterTagFailureEvent.java

```diff
@@ -7,7 +7,7 @@ public class AlterTagFailureEvent extends TagFailureEvent {
     private final String name;
     private final TagChange[] changes;
     public AlterTagFailureEvent(String user, String metalake, String name, TagChange[] changes, Exception exception) {
-        super(user, exception);
+        super(user, null, exception);
```

Comment:
please provide a meaning identifier to all failure event, for tag operations the resource identifier is the $metalake.$tagName, for list operations the the resource identifier is the objects which supports list tags, such as $metalake.$metadataObject in `listTagsForMetadataObject`

Reply:
Ok

Reply:
Do I need to replace all String metalake in the TagDispatcher interface, as well as the fields that have been replaced with NameIdentifier in the TagManager, with NameIdentifier and construct the NameIdentifier in operations instead?

Reply:
what do you mean about `Do I need to replace all String metalake in the TagDispatcher interface`?

Reply:
I mean, currently TagDispatcher's methods are passing metalake as a String, and then turn it to NameIdentifier in TagManager.
I'm asking about if I should turn it to NameIdentifier earlier in TagOperations and then pass it through event and manager.

Reply:
I perfer not changing the interfaces in TagManager

Reply:
Ok